### PR TITLE
Added TutorialActivity Test

### PR DESCRIPTION
**Pull request by Hiteshgautam01**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the checks with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by sagar15795** [comment]
Fix Travis issue

File: app/build.gradle [hunk 1]
```diff
@@ -126,6 +126,7 @@ dependencies {
 
     //Dependencies for JUNit and unit tests.
     testImplementation "junit:junit:4.12"
+
```
Comment (sagar15795) [inline app/build.gradle]: remove this line.

File: app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java [hunk 2]
```diff
@@ -102,20 +102,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
 
-
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java]: revert this line.

File: app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java [hunk 3]
```diff
@@ -102,20 +102,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
         mEditTextEmail.addTextChangedListener(new CustomTextWatcher(mEditTextEmail));
-
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java]: revert this line.

File: app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java [hunk 4]
```diff
@@ -102,20 +102,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
-
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java]: revert this line.

File: app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java [hunk 5]
```diff
@@ -102,20 +102,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
         mEditTextPassword.addTextChangedListener(new CustomTextWatcher(mEditTextPassword));
-
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java]: revert this line.

File: app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java [hunk 6]
```diff
@@ -102,20 +102,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
         mEditTextEmail.setOnFocusChangeListener(this);
-
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java]: revert this line.

File: app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java [hunk 7]
```diff
@@ -102,20 +102,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
         mEditTextPassword.setOnFocusChangeListener(this);
-
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java]: revert this line.

File: app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java [hunk 8]
```diff
@@ -102,20 +102,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
         progressDialog = new ProgressDialog(getContext());
-
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java]: revert this line.

File: app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java [hunk 9]
```diff
@@ -102,20 +102,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
         progressDialog.setMessage("Please wait");
-
```
Comment (sagar15795) [inline app/src/main/java/org/apache/taverna/mobile/ui/login/LoginFragment.java]: revert this line.

File: app/src/androidTest/java/org/apache/taverna/mobile/tutorial/TutorialActivityTest.java [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    public void clickingNext_shouldGotoNextTutorial_onClickGotIt_ShouldGoToLoginActivity()
+            throws Exception {
+
+        Intents.init();
+        onView(withText(R.string.next)).check(matches(isDisplayed()));
+        onView(withId(R.id.slide_pager)).perform(swipeLeft());
+        onView(withId(R.id.slide_pager)).perform(swipeLeft());
+        onView(withId(R.id.slide_pager)).perform(swipeLeft());
+        onView(withId(R.id.slide_pager)).perform(swipeLeft());
+        onView(withText("GOT IT")).check(matches(isDisplayed()));
```
Comment (sagar15795) [inline app/src/androidTest/java/org/apache/taverna/mobile/tutorial/TutorialActivityTest.java]: also verify that skip should not visible when "GOT IT" button is visible.

**Comment by Hiteshgautam01** [comment]
Hey @sagar15795, Done with the requested changes. Please Review.

**Comment by sagar15795** [comment]
@Hiteshgautam01 I have also request you to add  one more scenario in last comment. Please add it also.

File: app/src/androidTest/java/org/apache/taverna/mobile/tutorial/TutorialActivityTest.java [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+    public void clickingNext_shouldGotoNextTutorial_onClickGotIt_ShouldGoToLoginActivity()
+            throws Exception {
+
+        Intents.init();
+        onView(withText(R.string.next)).check(matches(isDisplayed()));
+        onView(withId(R.id.slide_pager)).perform(swipeLeft());
+        onView(withId(R.id.slide_pager)).perform(swipeLeft());
+        onView(withId(R.id.slide_pager)).perform(swipeLeft());
+        onView(withId(R.id.slide_pager)).perform(swipeLeft());
+        onView(withText("SKIP")).check(matches(not(isDisplayed())));
```
Comment (Hiteshgautam01) [inline app/src/androidTest/java/org/apache/taverna/mobile/tutorial/TutorialActivityTest.java]: @sagar15795, I already added the skip check whether it is displayed or not when GOT IT is displayed.

Comment (sagar15795) [reply app/src/androidTest/java/org/apache/taverna/mobile/tutorial/TutorialActivityTest.java]: @Hiteshgautam01  `Please change the requested changes. Could you also write a test for the scenario that you first navigate to a random page and then click skip and verify is it navigating to LoginActivity or not?`

Comment (sagar15795) [reply app/src/androidTest/java/org/apache/taverna/mobile/tutorial/TutorialActivityTest.java]: I have also requested to add a new scenario.

Comment (Hiteshgautam01) [reply app/src/androidTest/java/org/apache/taverna/mobile/tutorial/TutorialActivityTest.java]: @sagar15795  Done

**Comment by sagar15795** [comment]
@Hiteshgautam01 Thanks for the PR.

**Pull request merged**